### ISSUE TITLE
Error in "minishift delete" with Xhyve

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#2067]
`minishift delete` returning error 
```
Error deleting '/Users/lmohanty/.minishift/machines': remove /Users/lmohanty/.minishift/machines/minishift/root-volume.sparsebundle/bands/9c3: permission denied
```  
and the subsequent `minishift start` fails with below error.

```
$ minishift start --openshift-version v3.6.1
-- Starting profile 'minishift'
open /Users/lmohanty/.minishift/machines/minishift/config.json: no such file or directory
```

### General information

  * Minishift version: minishift v1.13.1+60379334
  * OS: macOS
  * Hypervisor: xhyve 

### Steps to reproduce

I am not sure what triggers this issue, may be it is related to recent updates in `high sierra`.

  1. `minishift delete`

### Expected
```
$ minishift delete --force
Deleting the Minishift VM...
Minishift VM deleted.
```
### Actual
```
$ minishift delete --force
Deleting the Minishift VM...
Error deleting '/Users/lmohanty/.minishift/machines': remove /Users/lmohanty/.minishift/machines/minishift/root-volume.sparsebundle/bands/9c3: permission denied
```

```
$ ls -al ~/.minishift/machines/minishift/root-volume.sparsebundle/bands/
total 35968
drwxr-xr-x  11 root      staff      352 Feb 26 12:21 .
drwxr-xr-x@  3 lmohanty  staff       96 Feb 26 12:23 ..
-rw-r--r--   1 root      staff  3264512 Feb 26 12:21 0
-rw-r--r--   1 root      staff   122880 Feb 26 12:20 19
-rw-r--r--   1 root      staff  7274496 Feb 26 12:22 2ce
-rw-r--r--   1 root      staff  8388608 Feb 26 12:23 520
-rw-r--r--   1 root      staff  7012352 Feb 26 12:22 521
-rw-r--r--   1 root      staff  2555904 Feb 26 12:22 772
-rw-r--r--   1 root      staff  1277952 Feb 26 12:23 7d
-rw-r--r--   1 root      staff  8368128 Feb 26 12:20 9b3
-rw-r--r--   1 root      staff  8388608 Feb 26 12:21 9c3
```

**Comment by LalatenduMohanty** [comment]
Workaround
===

```
sudo rm -rf ~/.minishift/machines
```

**Comment by gbraad** [comment]
Are you sure this is the workaround? as I believe tit is more related to the sparse filesystem mount and timing...

**Comment by LalatenduMohanty** [comment]
> Are you sure this is the workaround? as I believe tit is more related to the sparse filesystem mount and timing...

Yes, the workaround works for me. Though I am not sure why the permission error comes in the first place. Because after successful start the permissions are same in `~/.minishift/machines/minishift/root-volume.sparsebundle/bands/`

```
 ls -al machines/minishift/root-volume.sparsebundle/bands/ | grep 9c3
-rw-r--r--    1 root      staff  8388608 Feb 26 12:38 9c3
```

**Comment by praveenkumar** [comment]
Following what I get when `stop` the VM. Are you folks also experience same?

```
$ ./minishift stop
Stopping local OpenShift cluster...
Error stopping cluster: unexpected EOF
```

**Comment by LalatenduMohanty** [comment]
@praveenkumar I have not see the error with stop yet.

**Comment by amitkrout** [comment]
@praveenkumar yes i have faced this but not all the time

**Comment by amitkrout** [comment]
Adding to this https://github.com/minishift/minishift/issues/2067#issuecomment-368408533

This looks hard but works for me as a workaround 
```sudo minishift delete --force```

**Comment by LalatenduMohanty** [comment]
Resolved via #2097
